### PR TITLE
[FS-2178] iOS Config to Add System and Custom Frameworks

### PR DESCRIPTION
**Pull request by crherman7**
# Description

Ticket: https://brandingbrand.atlassian.net/browse/FS-2178

Add iOS config and logic to add system and custom frameworks to iOS project.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes i.e. STR. Please also list any relevant details for your test configuration.

- [x] Test A
1. `yarn prepare`
2. Copy the flagship directory from node modules to another project
3. Add ios.frameworks config to Flagship configuration
4. `yarn install`
5. `yarn run init`
6. Open up xCode and verify that the frameworks have been added

Screen Shot 2021-04-02 at 2 56 22 PM

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code only in work-arounds and 3rd party integration interfaces
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added test image/video that proves my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

# Example
```
  frameworks: [
    {
      framework: 'StoreKit.framework'
    },
    {
      framework: 'APIGuard.framework',
      frameworkPath: 'scripts/shape/ios'
    }
  ]
```

### CODE REVIEW

File: packages/flagship/src/lib/ios.ts [hunk 1]
```diff
@@ -495,3 +495,28 @@ export function setEnvSwitcherInitialEnv(configuration: Config, env: string): vo
...
   );
 }
+
+/**
+ * Adds iOS system and custom frameworks to project
+ * @param {object} configuration The project configuration.
+ */
+export function frameworks(configuration: Config): void {
+  var projectPath = path.ios.pbxprojFilePath(configuration);
+  var project = xcode.project(projectPath);
```
Comment (crherman7) [inline packages/flagship/src/lib/ios.ts:505]: change to `const`

Comment (crherman7) [reply packages/flagship/src/lib/ios.ts]: declaration chaged

**Comment by bweissbart** [comment]
@deemaabdallah - can you please do a quick sanity check here?

**Comment by crherman7** [comment]
Missing whitespace, will update....idk if my setup is missing a commit hook or something

**Review by skyeckstrom — DISMISSED** [review]

**Review by skyeckstrom — APPROVED** [review]

**Pull request merged**